raft-otel/README.md

```diff
@@ -0,0 +1,112 @@
+requests, why particular requests are outliers in terms of latency, and how a single user request flows through backend services - many complex web services
+may involve dozens of backend services or datastores. It may be possible to answer some of these questions using logs analysis. 
+However, there is a better solution, designed just for this problem: distributed tracing. 
+
+Distributed tracing has two key concepts: traces and spans. A trace represents a whole request or transaction. Traces are uniquely identified by trace IDs.
+Traces are made up of a set of spans, each tagged with the trace ID of the trace it belongs to. Each span is a unit of work: a remote procedure call or 
+web request to a specific service, a method execution, or perhaps the time that a message spends in a queue. Spans can have child spans. 
+There are specific tools that are designed to collect and store distributed traces, and to perform useful queries against them. 
+One of the key aspects of distributed tracing is that when services call other services the trace ID is propagated to those calls, so that the 
+overall trace may be assembled. 
+
+[OpenTelemetry](https://opentelemetry.io/) (also known as OTel) is the main industry standard for distributed tracing. It governs the format of traces and spans, and how traces and spans are collected. It is worth spending some time exploring the [OTel documentation](https://opentelemetry.io/docs/), particularly the Concepts
+section. The [awesome-opentelemetry repo](https://github.com/magsther/awesome-opentelemetry) is another very comprehensive set of 
+resources.
+
+Distributed tracing is a useful technique for understanding how complex systems such as RAFT are operating. The goal of this project is to use distributed
+tracing to trace and to visualise operations in an implementation of RAFT, which is a nontrivial distributed system.
+
+### Part 1: Get RAFT working
+
+To begin with, we will need a running implementation of RAFT. Eli Bendersky has written a 
+[detailed set of blog posts describing a RAFT implementation in Go](https://eli.thegreenplace.net/2020/implementing-raft-part-0-introduction/).
+Read these blog posts carefully. 
+
+You can either 
+ 1) try to write your own RAFT implementation, building up the functionality in the stages described by Bendersky, or
+ 2) use Bendersky's code, after having thoroughly read it and understood it
+ 3) write your own implementation, using Bendersky's as a reference if you get stuck
+
+Reading code written by others is a useful skill to have, so if you opt to create your own implementation, you should review Bendersky's code.
+Does it differ from yours in any significant respect?
+
+By the end of Part 1 we should have a running RAFT cluster with 5 instances. 
+We may choose to run our RAFT cluster locally using `docker-compose`, `minikube`, or any other appropriate tool.
+
```

Comment:
It's probably worth describing some "write a small client to actually use your RAFT server" piece here? (Either as something interactive to run, or as one or two large intergration-y tests). Particularly if we tie this to the concrete example use case I mentioned above - I suspect this will help the people doing the exercise solidify in their mind what they've just done, that it works, and think about what they should be tracing.

Reply:
done!